### PR TITLE
Standardize major names

### PR DESCRIPTION
**Pull request by cubetastic33**
NP-81

Also fetches list of supported majors directly from validator instead of using a majors.json file.

Couldn't figure out how to rename the `VALIDATOR` env var though, so ended up creating a duplicate env var for now called `NEXT_PUBLIC_VALIDATOR`.

### CODE REVIEW

**Review by akevinge — CHANGES_REQUESTED** [review]
PTAL.

- You'll also need to modify the env schema: https://github.com/UTDNebula/planner/blob/develop/src/env/schema.mjs
- Build workflow: https://github.com/UTDNebula/planner/blob/develop/.github/workflows/build.yml

Don't worry about the integration tests. Also will update this and Vercel envs.

**Comment by akevinge** [comment]
Ok just updated integration test to run the validator inside the workflow (it was using the deployed instance before). Please rebase and update VALIDATOR in this file as well: https://github.com/UTDNebula/planner/blob/develop/.github/workflows/tests.yml.

File: validator/degree_data/Visual Performing Arts Music Concentration(BA).json [hunk 1]
```diff
@@ -189,5 +188,7 @@
         }
       }
     ]
-  }
-}
+  },
+  "display_name": "Visual and Performing Arts with Music Concentration(BA)",
```
[the remark this answers is not in the document]
Comment (cubetastic33) [reply validator/degree_data/Visual Performing Arts Music Concentration(BA).json:191]: good point

File: validator/degree_data/Visual Performing Arts Music Concentration(BA).json [hunk 2]
```diff
@@ -189,5 +188,7 @@
...
+  "uuid": "afcfff81-d8e9-42bd-9e90-1b3a3c4925a5"
```
[the remark this answers is not in the document]
Comment (cubetastic33) [reply validator/degree_data/Visual Performing Arts Music Concentration(BA).json]: oh yeah

File: validator/scripts/match_degree_names.py
[the remark this answers is not in the document]
Comment (cubetastic33) [reply validator/scripts/match_degree_names.py:1]: no

File: validator/degree_data/Business Administration(BS)_notready.json
[the remark this answers is not in the document]
Comment (cubetastic33) [reply validator/degree_data/Business Administration(BS)_notready.json:1]: oh yeah

File: src/components/planner/EditableMajor.tsx [hunk 3]
```diff
@@ -63,7 +62,17 @@ const EditMajorAutocomplete = ({
   setEditMajor: (a: boolean) => void;
   planId: string;
 }) => {
-  const majors = majorsList as string[];
+  const [majors, setMajors] = useState<string[]>([]);
+
+  fetch(`${process.env.NEXT_PUBLIC_VALIDATOR}/get-degree-plans`, {
```
[the remark this answers is not in the document]
Comment (cubetastic33) [reply src/components/planner/EditableMajor.tsx:67]: good idea

File: validator/api.py [hunk 4]
```diff
@@ -81,6 +83,18 @@ def root_() -> Response:
     )
 
 
+@app.route("/get-degree-plans", methods=["GET"])
+def get_degree_plans() -> Response:
+    plans = []
+    for fname in glob("./degree_data/*.json"):
```
[the remark this answers is not in the document]
Comment (cubetastic33) [reply validator/api.py]: ok

**Review by akevinge — DISMISSED** [review]
It seems like most of the checks aren't running. Try rebasing.

Also, could you update `VALIDATOR` in the [`e2e test workflow`](https://github.com/UTDNebula/planner/blob/develop/.github/workflows/tests.yml.) and write comments explaining what the hook/endpoint does.

Other than that, lgtm.

**Comment by akevinge** [comment]
Will update the envs on Vercel. Having issues logging in :thinking:

**Review by akevinge — APPROVED** [review]

**Pull request merged**